# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: f2f431ee78a3
# Full Hash: f2f431ee78a35a577f8d89a469ab17c20f71e3f0
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-21 03:56:08
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 2: Make `TextEditor` and `HTMLEditor` implement `EditorBase::CollapseSelectionToEndOfLastLeafNode` by themselves r=m_kato
#   
#   It does different thing for `TextEditor` and `HTMLEditor`, and used almost
#   internally.  Therefore, it should be implemented in the sub classes and
#   we should name them better.
# ==============================================================================

diff -r edac234f1eba -r f2f431ee78a3 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Wed Sep 21 00:20:26 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Wed Sep 21 00:38:23 2022 +0000
@@ -158,6 +158,7 @@
 
   // EditorBase overrides
   MOZ_CAN_RUN_SCRIPT NS_IMETHOD BeginningOfDocument() final;
+  MOZ_CAN_RUN_SCRIPT NS_IMETHOD EndOfDocument() final;
 
   NS_IMETHOD GetDocumentCharacterSet(nsACString& aCharacterSet) final;
   MOZ_CAN_RUN_SCRIPT NS_IMETHOD
@@ -2685,6 +2686,13 @@
    */
   [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult InitEditorContentAndSelection();
 
+  /**
+   * Collapse `Selection` to the last leaf content of the <body> or the document
+   * element.
+   */
+  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
+  CollapseSelectionToEndOfLastLeafNodeOfDocument() const;
+
   MOZ_CAN_RUN_SCRIPT nsresult SelectAllInternal() final;
 
   [[nodiscard]] Element* ComputeEditingHostInternal(